torch/_functorch/_aot_autograd/traced_function_transforms.py

```diff
@@ -235,22 +236,45 @@ def inner_fn(primals: List[Any], tangents: List[Any]):
         backward_out: Tuple[Tensor, ...] = ()
         # Call the backwards pass
         if grad_primals:
-            with fx_traceback.preserve_node_meta():
-                # for full graph export, we always export a joint graph where we assume no tangents are needed.
-                if aot_config.no_tangents:
-                    assert len(needed_tangents) == 1 and needed_tangents[0].numel() == 1
-                    backward_out = torch.autograd.grad(
-                        needed_outs,
-                        grad_primals,
-                        allow_unused=True,
+            with contextlib.ExitStack() as context_stack:
```

Comment:
Why the switch to `contextlib.ExitStack()` over an explicit context manager?
```
with fx_traceback.preserve_node_meta(), set_partitioner_tag(...):
    ....
```

Reply:
Yeah, in the original version with saved values there were more control-logic, now can be simplified.